JBTM-3557: Migrate LRA booking quickstart from throntail to quarkus

mayankkunwar opened the pull request:
https://issues.redhat.com/browse/JBTM-3557

mayankkunwar commented:
The build failure is not related to the code changes in this pull request.

mayankkunwar commented:
The build failure is not related to the code changes in this pull request.

mmusgrov reviewed (changes requested):
There are still a number of references to thorntail and swarm in the quickstart that need cleaning up (`find rts/lra -type f -exec grep -l swarm {} \;` ) etc

I also ran the PR on my laptop and there are pending LRAs left in the object store (`tree rts/lra/ObjectStore`). After recovery the expectation is that all LRAs have recovered so the store should be empty. This may either be an error with the quickstart or with the PR but can you investigate.

I have updated the job to include the store in the list of saved job artifacts.

mayankkunwar commented on rts/lra/run.sh:
```diff
@@ -4,7 +4,6 @@ function finish {
 
     rm -rf $NARAYANA_INSTALL_LOCATION
 }
-trap finish EXIT
```
Here the finish function was called twice, 1st time when `finish` was called it killed all the running processes. And 2nd time `finish` was called when trapping EXIT, there were no process to kill, so the build was failing.

mmusgrov replied:
What happens if the script finishes before `finish` is called? Will the script still clean up after itself?

mayankkunwar replied:
Before script exits, everything is getting cleaned up only when the finish is called for first time [here](https://github.com/jbosstm/quickstart/blob/master/rts/lra/run.sh#L95).

mmusgrov replied:
No I mean that if you remove the trap call and the bash script fails before reaching the call to `finish` how do the java processes which were started get cleaned up?

mayankkunwar replied:
Okay, @mmusgrov I got your point, so instead of removing `trap` command, I will add few `if` conditions in the `finish` function

mmusgrov replied:
No. Just make sure the finish command doesn't try to kill the same process twice - ie make it idempotent.

mayankkunwar replied:
I am removing the `finish` call from the last line. The `finish` that is called by `trap` will do the cleanup. Currently the build will fail because before exiting the script, last command i.e. `curl` is failing due to the recovery issue.

And earlier the build was passing because the last command `finish` that was successfully executed before exiting the script.

mayankkunwar commented:
I have made changes as suggested by you @mmusgrov 
There is some problem with the recovery when I ran the original build with Thorntail. I think this problem was already there before I started working on this PR

mmusgrov commented:
> I have made changes as suggested by you @mmusgrov
> There is some problem with the recovery when I ran the original build with Thorntail. I think this problem was already there before I started working on this PR

{quote}
I also ran the PR on my laptop and there are pending LRAs left in the object store (tree rts/lra/ObjectStore). After recovery the expectation is that all LRAs have recovered so the store should be empty. This may either be an error with the quickstart or with the PR but can you investigate.

mmusgrov commented:
When I look at commit ff71b4c405cdd32259cbaafa9a1efc8818119d13 at github I get the error:

  `This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.`

and I don't see it when I perform a `git fetch` followed by a `git checkout pr/346` on my laptop.

Will you check how you added this extra commit.

mayankkunwar commented:
> {quote} I also ran the PR on my laptop and there are pending LRAs left in the object store (tree rts/lra/ObjectStore). After recovery the expectation is that all LRAs have recovered so the store should be empty. This may either be an error with the quickstart or with the PR but can you investigate.

Okay, I will investigate the problem

mayankkunwar commented:
> When I look at commit [ff71b4c](https://github.com/jbosstm/quickstart/commit/ff71b4c405cdd32259cbaafa9a1efc8818119d13) at github I get the error:
> 
> `This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.`
> 
> and I don't see it when I perform a `git fetch` followed by a `git checkout pr/346` on my laptop.
> 
> Will you check how you added this extra commit.

You are getting the error message because you are checking the commit on [jbosstm/quickstart](https://github.com/jbosstm/quickstart/commit/ff71b4c405cdd32259cbaafa9a1efc8818119d13) branch, but actually I committed these changes on [mayankkunwar/quickstart](https://github.com/mayankkunwar/quickstart/commit/ff71b4c405cdd32259cbaafa9a1efc8818119d13) branch

mmusgrov commented:
https://github.com/jbosstm/quickstart/blob/master/rts/lra/pom.xml#L59 needs removing

mayankkunwar commented:
> https://github.com/jbosstm/quickstart/blob/master/rts/lra/pom.xml#L59 needs removing

Sorry I missed this one

mmusgrov commented:
BTW I wouldn't push more commits until you have tracked down why there are still records in the object store even though each participant has been notified since it needlessly consumes CI resources.

Also, what is your plan for the unrelated quickstart failure? At a minimum it needs to be tracked in JIRA.

mmusgrov commented:
If you track the pom changes since the last successful run then the most obvious culprit is the version bump of h2.

mayankkunwar commented:
> BTW I wouldn't push more commits until you have tracked down why there are still records in the object store even though each participant has been notified since it needlessly consumes CI resources.
> 
> Also, what is your plan for the unrelated quickstart failure? At a minimum it needs to be tracked in JIRA.

Okay, sure @mmusgrov I will fix the recovery issue first. 
And for the unrelated quickstart failure I will open new issue on JIRA.

mayankkunwar commented:
> If you track the pom changes since the last successful run then the most obvious culprit is the version bump of h2 and the release notes for that version are https://github.com/h2database/h2database/releases/tag/version-2.0.202 (and http://www.h2database.com/html/changelog.html)

@mmusgrov I don't think bump of h2 version is causing the recovering problem. If you see the logs of last [successful build](http://narayanaci1.eng.hst.ams2.redhat.com/job/btny-pulls-narayana-quickstart/31/consoleFull) and search for keyword `flight cancel problem`, you will see that recovery problem exist before `quickstart/pull/340`

mmusgrov commented:
> @mmusgrov I don't think bump of h2 version is causing the recovering problem. If you see the logs of last [successful build](http://narayanaci1.eng.hst.ams2.redhat.com/job/btny-pulls-narayana-quickstart/31/consoleFull) and search for keyword `flight cancel problem`, you will see that recovery problem exists before `quickstart/pull/340`

Yes I know, my comment was related to the

> the unrelated quickstart failure ...

and my point was that If you track the pom changes since the last successful run then the most obvious culprit is the version bump of h2 and the release notes for that version ...

ie I was providing you with help on how to tackle/report the unrelated failure.

mayankkunwar commented:
Thanks @mmusgrov, I also observed that bump of h2 is creating issues in other modules

mmusgrov commented:
@mayankkunwar if the reason for the extra transaction logs left in the object store is not clear then how about raising a JBTM for it and then we can merge this PR. WDYT?

mayankkunwar commented:
> @mayankkunwar if the reason for the extra transaction logs left in the object store is not clear then how about raising a JBTM for it and then we can merge this PR. WDYT? We'd probably also need to disable this quickstart until we have an idea what's happening.

@mmusgrov I think doing it as separate JBTM would be good idea. I will raise another JIRA for the same and I will disable this quickstart. Thanks

mmusgrov commented on rts/pom.xml:
```diff
@@ -42,7 +42,7 @@
   <modules>
     <module>at</module>
-    <module>lra</module>
+<!--    <module>lra</module>-->
```
We need a comment that indicates why it's been disabled - a reference to the JBTM should be sufficient.

mayankkunwar replied:
okay done
